File: tests/fixtures/dart/sample.dart
import 'dart:async';
import 'package:flutter/material.dart';
import 'package:my_app/models/user.dart';
import '../relative.dart';

abstract class Repository {
  Future<User?> find(int id);
  Future<void> save(User user);
}

enum UserStatus { active, inactive }

class UserService extends BaseService implements Repository {
  final UserRepository _repo;

  UserService(this._repo);

  @override
  Future<User?> find(int id) async {
    return await _repo.findById(id);
  }

  @protected
  void _validate(User user) {
    if (user.name.isEmpty) throw ArgumentError('Empty name');
  }
}

mixin Loggable {
  void log(String msg) => print(msg);
}

Future<UserService> createService() async {
  return UserService(InMemoryRepo());
}

File: tests/fixtures/swift/Sample.swift
import Foundation
import UIKit

protocol Repository {
    func find(id: Int) -> User?
    func save(user: User)
}

@objc
open class UserService: BaseService, Repository {
    private let repo: UserRepository

    init(repo: UserRepository) {
        self.repo = repo
    }

    @discardableResult
    public func getUser(id: Int) async throws -> User {
        return try await repo.find(id: id)
    }

    internal func validate(_ user: User) -> Bool {
        return !user.name.isEmpty
    }
}

enum UserStatus {
    case active
    case inactive
}

struct UserDTO {
    let name: String
    let email: String
}

func createService() -> UserService {
    return UserService(repo: InMemoryRepo())
}

func classify(x: Int) -> String {
    if x > 0 {
        return "positive"
    } else if x < 0 {
        return "negative"
    }
    return "zero"
}
